### PR TITLE
(GH-2224) Add upgrade instructions for Cake 2.0

### LINKED ISSUE #2224: Upgrade instructions for Cake 2.0
We should add upgrade instructions for all breaking changes in Cake 2.0 to https://cakebuild.net/docs/getting-started/upgrade

### PR DESCRIPTION
**Pull request by pascalberger**
Adds upgrade instructions for Cake 2.0

Fixes #2224

### CODE REVIEW

File: input/docs/getting-started/upgrade.md [hunk 1]
```diff
@@ -44,6 +44,151 @@ To update Cake follow these instructions
...
+Cake will continue to support building of .NET Framework projects as well as projects targeting .NET Core 3.0 or older.
+
+Supported platform matrix for Cake 2.0 will look like this:
+
+| Runner                           | .NET 6 | .NET 5 | .NET Core 3.1 |
+| -------------------------------- |:------:|:------:|:-------------:|
+| [Cake .NET Tool]                 | <i class="fa fa-check" style="color:green"></i> | <i class="fa fa-check" style="color:green"></i> | <i class="fa fa-check" style="color:green"></i> |
+| [Cake Frosting]                  | <i class="fa fa-check" style="color:green"></i> | <i class="fa fa-check" style="color:green"></i> | <i class="fa fa-check" style="color:green"></i> |
+
+As a consequence of no longer shipping [Cake runner for .NET Framework] we will also stop shipping the [Cake.Portable Chocolatey package]
```
Comment (gep13) [inline input/docs/getting-started/upgrade.md:69]: @pascalberger @augustoproiete Am I right in saying that these (the Chocolatey and Brew) packages won't be pushed out as part of the RC release?

File: input/docs/getting-started/upgrade.md [hunk 2]
```diff
@@ -44,6 +44,151 @@ To update Cake follow these instructions
...
+and [Homebrew Cake formulae].
+
+[Cake runner for .NET Core] has been deprecated since version 1.0 with [Cake .NET Tool] as the suggested replacement.
+For users of [Cake runner for .NET Framework] it is also suggested to switch to [Cake .NET Tool] and run builds on .NET Core 3.1 or newer.
+For users which rely on extension or other dependencies which require .NET Framework or .NET Core 3.0 or older, suggestion is to stay on Cake 1.x.
```
Comment (gep13) [inline input/docs/getting-started/upgrade.md:74]:
```suggestion
For users which rely on an extension, or other dependencies, which require .NET Framework or .NET Core 3.0 or older, suggestion is to stay on Cake 1.x.
```

File: input/docs/getting-started/upgrade.md [hunk 3]
```diff
@@ -44,6 +44,151 @@ To update Cake follow these instructions
...
+
+See [Sunsetting of .NET Framework and .NET Core runners in Cake 2.0] for details.
+
+## Cake Frosting
+
+The deprecated `ReverseDependencyAttribute` and `DependencyAttribute` have been removed.
+`ReverseDependencyAttribute` can be replaced with `IsDependeeOffAttribute` and `DependencyAttribute` with `IsDependentOnAttribute`.
```
Comment (gep13) [inline input/docs/getting-started/upgrade.md:81]:
```suggestion
`ReverseDependencyAttribute` can be replaced with `IsDependeeOfAttribute` and `DependencyAttribute` with `IsDependentOnAttribute`.
```

File: input/docs/getting-started/upgrade.md [hunk 4]
```diff
@@ -44,6 +44,151 @@ To update Cake follow these instructions
...
+
+The data type of `AzurePipelinesBuildInfo.ArtifactStagingDirectory`, `AzurePipelinesBuildInfo.BinariesDirectory`, `AzurePipelinesBuildInfo.SourcesDirectory`,
+`AzurePipelinesBuildInfo.StagingDirectory` and `AzurePipelinesBuildInfo.TestResultsDirectory` has been changed from `FilePath` to `DirectoryPath`.
+
+See [#3590](https://github.com/cake-build/cake/issues/3590) for details.
+
+### Removal of TFBuildProvider
+
+Deprecated `TFBuildProvider` has been removed.
+It can be replaced with `AzurePipelinsProvider`.
```
Comment (gep13) [inline input/docs/getting-started/upgrade.md:124]:
```suggestion
It can be replaced with `AzurePipelinesProvider`.
```

**Review by gep13 — CHANGES_REQUESTED** [review]
Just a few small minor suggestions.

File: input/docs/getting-started/upgrade.md (hunk 1 above)
Comment (pascalberger) [reply input/docs/getting-started/upgrade.md:69]: Yes, we no longer will push to Chocolatey and Brew

**Comment by pascalberger** [comment]
@gep13 Fixed the issues and ready for another review

**Review by gep13 — APPROVED** [review]
LGTM!

**Pull request merged**